### PR TITLE
Typify internal list in DirectGraphLayout and convert to Deque

### PR DESCRIPTION
**Pull request by ptziegler**
By using a deque, we have access to both an iterator() and an descendingIterator() method. This allows us to iterate over the collection in both directions.

### CODE REVIEW

**Comment by ptziegler** [comment]
#155 

Switching to a Deque should be safe, given that this field is a package-private class.

File: org.eclipse.zest.layouts/src/org/eclipse/zest/layouts/algorithms/DirectedGraphLayoutAlgorithm.java [hunk 1]
```diff
@@ -34,20 +34,14 @@ public void visit(DirectedGraph graph) {
 				field = DirectedGraphLayout.class.getDeclaredField("steps");
```
Comment (azoitl) [inline org.eclipse.zest.layouts/src/org/eclipse/zest/layouts/algorithms/DirectedGraphLayoutAlgorithm.java:34]: should we add a protected getter to the steps so that we don't need reflection here?

Comment (ptziegler) [reply org.eclipse.zest.layouts/src/org/eclipse/zest/layouts/algorithms/DirectedGraphLayoutAlgorithm.java:34]: I'm not sure if that would really work... "steps" stores objects of type GraphVisitor, which is a package-private class in draw2d. With a getter, this class would now be exposed.
As an example, the method call in DirectedGraphLayoutAlgorithm would still have to look like `Deque<?> parts = getParts()`, simply because GraphVisitor is not visible.

Comment (azoitl) [reply org.eclipse.zest.layouts/src/org/eclipse/zest/layouts/algorithms/DirectedGraphLayoutAlgorithm.java:34]: I meant to add a protected getSteps method to DirectedGraph Layout so that Children like this ExtenddedDirectGraphLayout could access the steps. But I'm not sure if we want to expose it like that. As you say this has implications.

**Comment by azoitl** [comment]
@ptziegler have you seen my latest comment? It would be great if we could merge that commit somewhere today or early tomorrow as we have M3 deadline.

File: org.eclipse.zest.layouts/src/org/eclipse/zest/layouts/algorithms/DirectedGraphLayoutAlgorithm.java (hunk 1 above)
Comment (ptziegler) [reply org.eclipse.zest.layouts/src/org/eclipse/zest/layouts/algorithms/DirectedGraphLayoutAlgorithm.java:34]: Let me rephrase it: What should getSteps() return? It can't be Deque<GraphVisitor> because the GraphVisitor is package-private class. So the only other alternatives are Deque<?> or Deque<Object>, both of which I'm not really happy with...

Comment (azoitl) [reply org.eclipse.zest.layouts/src/org/eclipse/zest/layouts/algorithms/DirectedGraphLayoutAlgorithm.java:34]: Ah now I get it. Yes a getter makes here then no sense at all.

**Comment by azoitl** [comment]
No problem. Thanks for the feedback. Will merge it now.
